### PR TITLE
fix: generate legal json for keys that contain quote

### PR DESCRIPTION
**Pull request by virgilp**
Bugfix for self-explanatory bug in `JsonLayout` (see test)

### CODE REVIEW

File: src/NLog/Layouts/JSON/JsonLayout.cs [hunk 1]
```diff
@@ -333,7 +333,7 @@ private void BeginJsonProperty(StringBuilder sb, string propName, bool beginJson
             }
 
             sb.Append('"');
-            sb.Append(propName);
+            sb.Append(propName.Replace(@"""",@"\"""));
```
Comment (snakefoot) [inline src/NLog/Layouts/JSON/JsonLayout.cs:336]: Could you do this instead: `Targets.DefaultJsonSerializer.AppendStringEscape(sb, propName, false, false);`

Comment (virgilp) [reply src/NLog/Layouts/JSON/JsonLayout.cs]: Done

**Comment by snakefoot** [comment]
@virgilp  I you want this included in NLog 4.7.7 then I guess you should re-base to master-branch.

**Comment by virgilp** [comment]
I cherry-picked the change and created a new PR against master - is that ok? 
https://github.com/NLog/NLog/pull/4202

**Comment by snakefoot** [comment]
> I cherry-picked the change and created a new PR against master - is that ok? #4202

Yes it looks good. You can close this PR.

**Pull request closed without merging**